apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/service/state/echange/impl/ExchangeServiceImpl.java

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright © 2018-2020 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.core.service.state.echange.impl;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import com.apollocurrency.aplwallet.apl.core.app.Block;
+import com.apollocurrency.aplwallet.apl.core.app.Transaction;
+import com.apollocurrency.aplwallet.apl.core.converter.rest.IteratorToStreamConverter;
+import com.apollocurrency.aplwallet.apl.core.dao.state.exchange.ExchangeTable;
+import com.apollocurrency.aplwallet.apl.core.db.DbClause;
+import com.apollocurrency.aplwallet.apl.core.db.DbIterator;
+import com.apollocurrency.aplwallet.apl.core.entity.state.exchange.Exchange;
+import com.apollocurrency.aplwallet.apl.core.monetary.CurrencyExchangeOffer;
+import com.apollocurrency.aplwallet.apl.core.service.state.BlockChainInfoService;
+import com.apollocurrency.aplwallet.apl.core.service.state.echange.ExchangeService;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Singleton
+public class ExchangeServiceImpl implements ExchangeService {
+
+    private final ExchangeTable exchangeTable;
+    private final BlockChainInfoService blockChainInfoService;
+    private final IteratorToStreamConverter<Exchange> exchangeIteratorToStreamConverter;
+
+    @Inject
+    public ExchangeServiceImpl(ExchangeTable exchangeTable, BlockChainInfoService blockChainInfoService) {
+        this.exchangeTable = exchangeTable;
+        this.blockChainInfoService = blockChainInfoService;
+        this.exchangeIteratorToStreamConverter = new IteratorToStreamConverter<>();
+    }
+
+    /**
+     * for unit tests
+     */
+    public ExchangeServiceImpl(ExchangeTable exchangeTable,
+                               BlockChainInfoService blockChainInfoService,
+                               IteratorToStreamConverter<Exchange> exchangeIteratorToStreamConverter) {
+        this.exchangeTable = exchangeTable;
+        this.blockChainInfoService = blockChainInfoService;
+        if (exchangeIteratorToStreamConverter != null) {
```

Comment:
do we really need this `if` here? Why not just set in a unit test?

Reply:
I've put that just to create 'fully prepared' compoonent in any case, so we don't have inconsistent component.